experimental/casemapping/src/casemapping.rs

```diff
@@ -49,9 +59,24 @@ impl CaseMapping {
 
     /// A constructor which creates a [`CaseMapping`] for the given locale.
     ///
+    /// This does not perform any complex fallback: the locale used must have an exact match
```

Comment:
thought: this wouldn't be hard to add, we can just go through the fallback chain until something non-rooty appears. Actually it's weird that this uses `&Locale`, all other constructors use `&DataLocale`.

Reply:
ICU4C does this as well and it makes sense: you kinda want exact matches here as long as it's documented clearly. The region should not affect the casemapping algorithm, the language should. 

We're not loading data here based on this, so it shouldn't be a data locale. Could be a language though.

Reply:
> We're not loading data here based on this, so it shouldn't be a data locale.

This is an implementation detail.

Reply:
I don't really think it is; it's a different kind of locale parameter in my book.

Reply:
Alternatively, we could move the parameter to the methods, as it's not needed in the constructor. There are already special `_turkic` methods, which we could replace by more general `_for_locale` methods.

Reply:
I'm fine with that. I don't think it needs to be on the constructor and was considering bringing it up as a part of final API polish (I've already listed a bunch of other things on the related issue)

We can expose the enum and also a From/Default.

How about we merge this as-is and I do the rest in a separate PR

Reply:
Sure, this is not blocking.

The `Greek` and `Dutch` variants are actually unused.

Reply:
Yeah there's some Greek special casing that we don't yet support

Reply:
> There are already special `_turkic` methods, which we could replace by more general `_for_locale` methods.

We should really not do that. The Turkic case _foldings_ need to be handled with extreme care, and must not be confused with the actual locale-dependence of case _mappings_.

See Markus’s comments on https://github.com/unicode-org/icu4x/issues/3151#issuecomment-1452524811.

Reply:
Ah I hadn't seen that. Continuing the discussion on #3234